crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -2346,15 +2346,13 @@ impl CheckpointService {
         tasks.spawn(monitored_future!(builder.run()));
         tasks.spawn(monitored_future!(aggregator.run()));
 
-        loop {
-            if tokio::time::timeout(Duration::from_secs(10), self.wait_for_rebuilt_checkpoints())
-                .await
-                .is_ok()
-            {
-                break;
-            } else {
-                debug_fatal!("Still waiting for checkpoints to be rebuilt");
-            }
+        // If this times out, the validator may still start up. The worst that can
+        // happen is that we will crash later on (due to missing transactions).
+        if tokio::time::timeout(Duration::from_secs(10), self.wait_for_rebuilt_checkpoints())
```

Comment:
just double checking, sure 10s is way more time than we should ever need for this, or would it make sense to bump it to 30 or 60 or something without the loop?

Reply:
since we only crash in tests, and we will catch the debug_fatal with monitoring, i think we can just bump up the time later if needed?